packages/ember-runtime/lib/mixins/array.js

```diff
@@ -142,17 +142,23 @@ export function arrayContentDidChange(array, startIdx, removeAmt, addAmt) {
   let cache = meta && meta.readableCache();
 
   if (cache) {
+    let length = get(array, 'length') + removeAmt - addAmt;
     if (cache.firstObject !== undefined &&
-        objectAt(array, 0) !== cacheFor.get(cache, 'firstObject')) {
+        ((startIdx === 0 && (addAmt > 0 || removeAmt > 0)) ||
```

Comment:
should we still include `objectAt(array, 0) !== cacheFor.get(cache, 'firstObject')` guard?